tests/tests_suite_test.go

```diff
@@ -35,26 +37,59 @@ var _ = AfterSuite(func() {
 	removeTestNamespaces()
 })
 
+func getPodLogs(podName, containerName string) (string, error) {
+	req := testClient.VirtClient.CoreV1().Pods(managerNamespace).GetLogs(podName, &corev1.PodLogOptions{
+		Container: containerName,
+	})
+	podLogs, err := req.Stream(context.TODO())
+	if err != nil {
+		return "", err
+	}
+	defer func(podLogs io.ReadCloser) {
+		err := podLogs.Close()
+		if err != nil {
+			panic(err)
+		}
+	}(podLogs)
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "", err
+	}
+	output := buf.String()
+
+	return output, nil
+}
+
 func KubemacPoolFailedFunction(message string, callerSkip ...int) {
-	podList, err := testClient.VirtClient.CoreV1().Pods(managerNamespace).List(context.TODO(), metav1.ListOptions{})
+	podList, err := testClient.VirtClient.CoreV1().Pods(managerNamespace).List(context.TODO(),
+		metav1.ListOptions{LabelSelector: "app=kubemacpool"})
 	if err != nil {
 		fmt.Println(err)
 		Fail(message, callerSkip...)
 	}
 
 	for _, pod := range podList.Items {
 		podYaml, err := testClient.VirtClient.CoreV1().Pods(managerNamespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
-
-		req := testClient.VirtClient.CoreV1().Pods(managerNamespace).GetLogs(pod.Name, &corev1.PodLogOptions{})
-		output, err := req.DoRaw(context.TODO())
 		if err != nil {
 			fmt.Println(err)
 			Fail(message, callerSkip...)
 		}
 
-		fmt.Printf("Pod Name: %s \n", pod.Name)
-		fmt.Printf("%v \n", *podYaml)
-		fmt.Println(string(output))
+		fmt.Printf("\nPod Name: %s \n", pod.Name)
+		fmt.Printf("Pod Yaml:\n%v \n", *podYaml)
+
+		for i := range pod.Spec.Containers {
+			containerName := pod.Spec.Containers[i].Name
+			podLogs, err := getPodLogs(pod.Name, containerName)
+			if err != nil {
+				fmt.Println(err)
+				Fail(message, callerSkip...)
+			}
+
+			fmt.Printf("\nPod container %q Logs:\n%s \n", containerName, podLogs)
+		}
```

Comment:
please consider to extract to a func

Reply:
including the for loop?

Reply:
yep i think so, this is the core of the helper

Reply:
DONE